src/include/duckdb/storage/table/row_group_collection.hpp

```diff
@@ -154,6 +155,25 @@ class RowGroupCollection {
 	}
 	void SetAppendRequiresNewRowGroup();
 
+	std::vector<idx_t> prefix_sums;
```

Comment:
We shouldn't be storing this in the `RowGroupCollection` - because this can be accessed in parallel by different threads (that might see different transaction versions of the same collection). This can also be done in parallel. This should be stored in the table state.